### DISCUSSION TITLE
How to upgrade to WSL2

### DISCUSSION THREAD

**Discussion by okarpov** [microsoft/WSL#9967] [Q&A]
How to upgrade from WSL to WSL2?

i have an issue and i need to use WSL2 to be able to mount a VHD drive
please, assist
`WslRegisterDistribution failed with error: 0x80370114
Error: 0x80370114 The operation could not be started because a required feature is not installed.`

`wsl --mount \\.\PhysicalDrive3 --bare This operation is only supported by WSL2. Error code: Wsl/Service/AttachDisk/WSL_E_WSL2_NEEDED`

`wsl --status Default Version: 2 WSL1 is not supported with your current machine configuration. Please enable the "Windows Subsystem for Linux" optional component to use WSL1.`

`wsl.exe --version
WSL version: 1.2.0.0
Kernel version: 5.15.90.1
WSLg version: 1.0.51
MSRDC version: 1.2.3770
Direct3D version: 1.608.2-61064218
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22621.1555`

**Comment by Biswa96** [comment]
What is the output of `wsl.exe -l -v` command?

**Reply by okarpov** [reply]
wsl.exe -l -v
Windows Subsystem for Linux has no installed distributions.

Use 'wsl.exe --list --online' to list available distributions
and 'wsl.exe --install <Distro>' to install.

Distributions can also be installed by visiting the Microsoft Store:
https://aka.ms/wslstore
Error code: Wsl/WSL_E_DEFAULT_DISTRO_NOT_FOUND

**Reply by Biswa96** [reply]
There is no Linux distribution installed in WSL. First install a Linux distribution of your choice.

**Reply by okarpov** [reply]
i did it many times but it does not help at all

**Reply by okarpov** [reply]
and to be able to install distro i need to set wsl default version to 1
but i need wsl2 to be able to add VHD disk to Linux

**Reply by Biswa96** [reply]
> and to be able to install distro i need to set wsl default version to 1

It is not required to be version 1. In the output, the default version is already 2.

**Reply by okarpov** [reply]
with default version set to 2:

Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80370114
Error: 0x80370114 The operation could not be started because a required feature is not installed.

Press any key to continue...

**Reply by okarpov** [reply]
re-read my original post, please.
It says i have WSL2 and that I can not have WSL1

but --version shows i have actually WSL1

that is why i'm asking how to upgrade to WSL 2

**Reply by Biswa96** [reply]
There are some issues reported with that error code https://github.com/microsoft/WSL/issues?q=is%3Aissue+is%3Aopen+0x80370114. Probably, the hyper-v and/or virtual machine platform optional features have to be installed.

**Reply by okarpov** [reply]
so you do not know how to upgrade to WSL2, right?

**Reply by Biswa96** [reply]
There are many variables. There WSL is installed in your system but some features are not enabled.
1. The `wsl.exe --update` command can retry to install those features.
2. Or, those features have to be manually enabled in optionalfeatures.exe window.
3. Or, virtualization has to be enabled.

There are many solutions provided in this issue https://github.com/microsoft/WSL/issues/9338

**Reply by okarpov** [reply]
i need the only one solution which will cause wsl --version to show 2.x.x.x and not as currently 1.x.x.x

i did try all those features and updates, nothing help

**Comment by tmax22** [comment]
I have the same issue. I have 2 users on my local machine.`user1` and `user2` both are admins.
I want to use the WSL distro created from user1 from user2, so i tried to mount the disk onto `user2`, so i ran this command  from `user2`:
```
❯ wsl --mount C:\Users\user1\AppData\Local\Packages\CanonicalGroupLimited.Ubuntu_79rhkp1fndgsc\LocalState\ext4.vhdx  ─╯
This operation is only supported by WSL2.
Error code: Wsl/Service/AttachDisk/WSL_E_WSL2_NEEDED
```

how do I effectively import the WSL disk from user1 to user2?
